dev/AnimatedVisualPlayer/AnimatedVisualPlayer.h

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "pch.h"
+#include "common.h"
+
+#include "AnimatedVisualPlayer.g.h"
+#include <winrt\Windows.UI.Xaml.Controls.h>
+
+// Derive from DeriveFromPanelHelper_base so that we get access to Children collection
+// in Panel. The Children collection holds the fallback content.
+struct AnimatedVisualPlayer final 
+    : ReferenceTracker<AnimatedVisualPlayer, DeriveFromPanelHelper_base, winrt::AnimatedVisualPlayer>
+{
+    AnimatedVisualPlayer();
+    ~AnimatedVisualPlayer();
+    winrt::IInspectable Diagnostics();
+    winrt::TimeSpan Duration();
+    winrt::IAnimatedVisualSource Source();
+    void Source(winrt::IAnimatedVisualSource const& value);
+    winrt::DataTemplate FallbackContent();
+    void FallbackContent(winrt::DataTemplate const& value);
+    bool AutoPlay();
+    void AutoPlay(bool value);
+    bool IsAnimatedVisualLoaded();
+    bool IsPlaying();
+    double PlaybackRate();
+    void PlaybackRate(double value);
+    winrt::Composition::CompositionObject ProgressObject();
+    ::winrt::Windows::UI::Xaml::Media::Stretch Stretch();
+    void Stretch(::winrt::Windows::UI::Xaml::Media::Stretch const& value);
+    void Pause();
+    winrt::IAsyncAction PlayAsync(double fromProgress, double toProgress, bool looped);
+    void Resume();
+    void SetProgress(double progress);
+    void Stop();
+
+    // FrameworkElement overrides
+    winrt::Size MeasureOverride(winrt::Size const& availableSize);
+    winrt::Size ArrangeOverride(winrt::Size const& finalSize);
+
+    // IUIElement / IUIElementOverridesHelper
+    winrt::AutomationPeer OnCreateAutomationPeer();
+
+    static void ClearProperties();
+    static void EnsureProperties();
+
+    // Dependency properties
+    static winrt::DependencyProperty AutoPlayProperty() { return s_AutoPlayProperty; }
+    static winrt::DependencyProperty DiagnosticsProperty() { return s_DiagnosticsProperty; }
```

Comment:
remove these dependency property related code and inherit this class from codegen property class for AnimatedViualPlayer.